pcap/pcap.h

```diff
@@ -837,7 +840,7 @@ PCAP_API pcap_t	*pcap_open(const char *source, int snaplen, int flags,
 PCAP_API int	pcap_createsrcstr(char *source, int type, const char *host,
 	    const char *port, const char *name, char *errbuf);
 PCAP_API int	pcap_parsesrcstr(const char *source, int *type, char *host,
-	    char *port, char *name, char *errbuf);
+	    char *port, char *name, unsigned char *uses_ssl, char *errbuf);
```

Comment:
`pcap_parsesrcstr()` is an API, so you can't change its calling sequence.

You would have to add a new `pcap_parsesrcstr_ex()` routine, with the new API, and make `pcap_parsesrcstr()` call it, passing a pointer to to internal variable for `uses_ssl`, and fail if that variable is non-zero when `pcap_parsesrcstr_ex()` returns.  Then have `pcap_open_rpcap()`, and other internal routines, call `pcap_parsesrcstr_ex()`.

Reply:
Also, there's no real benefit to having the Boolean flag be one byte long; the 3 bytes making it an `int` won't cost much, there might be padding causing it to eat all 4 bytes anyway, and some processors might do less internal work updating a 4-byte quantity than updating a 1-byte quantity.

Reply:
I reverted the change to the signature of `pcap_parsesrcstr()` and introduced a `pcap_parsesrcstr_ex()` that I call in the only place where it's really needed. For other call points, `pcap_parsesrcstr()` merely call the new function with a `NULL` as the ssl flag location.

> there's no real benefit to having the Boolean flag be one byte long

The actual benefit is that the only place where this pointer is needed is to initialize the flag within the `struct pcap_rpcap`, so making this function parameter a pointer to int instead would mean either introducing a temporary value (aka source code noise) or making the flag an int, but uint8 in `struct pcap_rpcap` seems the norm.

WDYT?